tests/GherkinTest.php

```diff
@@ -237,6 +239,20 @@ public static function resourceLineFilterDataProvider(): iterable
         ];
     }
 
+    public function testThatFileMustBeReadable(): void
+    {
+        $gherkin = new class($this->createMock(ParserInterface::class)) extends GherkinFileLoader {
+            public function parseInexistentFile(): void
+            {
+                $this->parseFeature('inexistent-file');
+            }
+        };
+
+        $this->expectExceptionObject(new RuntimeException('Cannot load feature from file; file could not be read: inexistent-file'));
+
+        $gherkin->parseInexistentFile();
```

Comment:
There is no reason to create a class extending `GherkinFileLoader`. Instead, use the actual GherkinFileLoader  class and call `$gherkin->parseFeature('inexistent-file')` in the test.